resources/assets/js/components/JobBuilder/JobBuilderRoot.tsx

```diff
@@ -13,47 +14,110 @@ import JobSkillsPage from "../JobBuilderSkills/JobBuilderSkillsPage";
 import JobReviewPage from "../JobReview/JobReviewPage";
 import ScrollToTop from "../ScrollToTop";
 
-const routes: Routes<any, React.ReactElement> = [
+const titles = defineMessages({
+  introTitle: {
+    id: "jobBuilder.intro.documentTitle",
+    defaultMessage: "Job Builder: Intro",
+    description: "The document's title shown in browser's title bar or tab.",
+  },
+  detailsTitle: {
+    id: "jobBuilder.details.documentTitle",
+    defaultMessage: "Job Builder: Details",
+    description: "The document's title shown in browser's title bar or tab.",
+  },
+  workEnvTitle: {
+    id: "jobBuilder.workEnv.documentTitle",
+    defaultMessage: "Job Builder: Work Environment",
+    description: "The document's title shown in browser's title bar or tab.",
+  },
+  impactTitle: {
+    id: "jobBuilder.impact.documentTitle",
+    defaultMessage: "Job Builder: Impact",
+    description: "The document's title shown in browser's title bar or tab.",
+  },
+  tasksTitle: {
+    id: "jobBuilder.tasks.documentTitle",
+    defaultMessage: "Job Builder: Tasks",
+    description: "The document's title shown in browser's title bar or tab.",
+  },
+  skillsTitle: {
+    id: "jobBuilder.skills.documentTitle",
+    defaultMessage: "Job Builder: Skills",
+    description: "The document's title shown in browser's title bar or tab.",
+  },
+  reviewTitle: {
+    id: "jobBuilder.review.documentTitle",
+    defaultMessage: "Job Builder: Review",
+    description: "The document's title shown in browser's title bar or tab.",
+  },
+});
+
+const routes: Routes<any, any> = [
   {
     path: "/:locale/manager/jobs",
     children: [
       {
         path: "/builder",
-        action: () => <JobBuilderIntroPageContainer jobId={null} />,
-      },
-      {
-        path: "/:id/builder/intro",
-        action: ({ params }) => (
-          <JobBuilderIntroPageContainer jobId={Number(params.id)} />
-        ),
-      },
-      {
-        path: "/:id/builder/details",
-        action: ({ params }) => <JobDetailsPage jobId={Number(params.id)} />,
-      },
-      {
-        path: "/:id/builder/environment",
-        action: ({ params }) => (
-          <JobBuilderWorkEnvPage jobId={Number(params.id)} />
-        ),
-      },
-      {
-        path: "/:id/builder/impact",
-        action: ({ params }) => (
-          <JobBuilderImpactPage jobId={Number(params.id)} />
-        ),
-      },
-      {
-        path: "/:id/builder/tasks",
-        action: ({ params }) => <JobTasksPage jobId={Number(params.id)} />,
-      },
-      {
-        path: "/:id/builder/skills",
-        action: ({ params }) => <JobSkillsPage jobId={Number(params.id)} />,
+        action: () => ({
+          title: <FormattedMessage {...titles.introTitle} />,
```

Comment:
`FormattedMessage` is a ReactElement that can only translate its contents if it's below [IntlProvider](https://github.com/formatjs/react-intl/blob/master/docs/Components.md#intlprovider) in the React element tree. It won't be much help for setting document.title. Instead, you should call intl.formatMessage to convert it directly to a string. You could change `routes` from a constant to a function that accepts the intl object, which JobBuilderRoot can pass in.

Reply:
You can use the [new hook](https://github.com/formatjs/react-intl/blob/master/docs/API.md#useintl-hook) to grab the intl object now